custom_components/tuya_local/devices/agl_ultramagic_lock.yaml

```diff
@@ -1,99 +1,56 @@
-name: AGL Ultra Magic lock
+name: AGL Ultra Magic Lock
+products:
+  - id: xrqkoapmq1rwjvms
+    name: Ultra Magic
 primary_entity:
-  entity: lock
+  entity: button
```

Comment:
I think it should be possible to combine the open and close using constraint and conditions into either a lock or cover entity.
```yaml
entity: cover
class: gate
dps:
  - id: 118
    type: boolean
    name: control
    mapping:
       - dps_val: true
          value: "open"
       - constraint: close_ctrl
          conditions:
            - dps_val: true
              value: "close"
  - id: 119
    type: boolean
    name: close_ctrl
    hidden: true
  - id: 3
    type: string
    name: open
    mapping:
      - dps_val: true
        value: false
      - dps_val: false
         value: true
```

Lock is maybe more difficult due to the separate dp 117 to indicate current state, and 118 and 119 to lock and unlock.